unicorn-core/src/main/scala/org/virtuslab/unicorn/repositories/Repositories.scala

```diff
@@ -2,12 +2,21 @@ package org.virtuslab.unicorn.repositories
 
 import org.virtuslab.unicorn.{ HasJdbcDriver, Tables, Identifiers }
 
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
 protected[unicorn] trait Repositories[Underlying]
     extends JunctionRepositories[Underlying]
     with IdRepositories[Underlying] {
   self: HasJdbcDriver with Identifiers[Underlying] with Tables[Underlying] =>
 
-  import driver.simple._
+  import driver.api._
+
+  protected val queryTimeout = 1.second
+
+  private[unicorn] def invokeAction[R, S <: slick.dbio.NoStream, E <: slick.dbio.Effect](action: DBIOAction[R, S, E])(implicit session: Session) = {
```

Comment:
def invokeAction[R, S <: slick.dbio.NoStream, E <: slick.dbio.Effect](action: DBIOAction[R, S, E])(implicit session: Session): R

It's basically the result type of invoked action